plugin/src/main/scala/org/ehcache/integrations/play/WrappedValueWithExpiry.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.integrations.play
+
+import org.ehcache.ValueSupplier
+import org.ehcache.expiry.{Expiry, Duration => EhDuration}
+
+import scala.concurrent.duration.Duration
+
+/**
+  * WrappedValueWithExpiry
+  */
+case class WrappedValueWithExpiry(value: Any, expiration: Duration) {
+  require(expiration.isFinite())
+}
+
+/**
+  * WrappedValueWithExpiryExpiration
+  */
+class WrappedValueWithExpiryExpiration extends Expiry[String, Any] {
```

Comment:
Isn't this just equivalent to the delegating class with a delegate of `Expirations.noExpiry`?